libpod/image/pull_test.go

```diff
@@ -243,12 +299,19 @@ func TestPullGoalFromImageReference(t *testing.T) {
 		if len(c.expected) == 0 {
 			assert.Error(t, err, c.srcName)
 		} else {
+			if c.mustFail {
+				require.Error(t, err, c.srcName)
+				continue
+			}
 			require.NoError(t, err, c.srcName)
 			require.Len(t, res.refPairs, len(c.expected), c.srcName)
 			for i, e := range c.expected {
 				testDescription := fmt.Sprintf("%s #%d", c.srcName, i)
 				assert.Equal(t, e.image, res.refPairs[i].image, testDescription)
-				assert.Equal(t, srcRef, res.refPairs[i].srcRef, testDescription)
+				if c.compareSourceRefs {
+					// They're not equal in case of a docker-archive.
+					assert.Equal(t, srcRef, res.refPairs[i].srcRef, testDescription)
```

Comment:
This fails just because it is comparing pointers for equality, doesn’t it? Could this instead compare `transports.ImageName(…)` values, allowing us to get rid of the `compareSourceRefs` option?

Reply:
Much better!